v0/destinations/wootric/util.js

```diff
@@ -0,0 +1,321 @@
+    const endpoint = `${BASE_ENDPOINT}/${VERSION}/end_users/phone_number/${phone}`;
+    return await userLookupResponseBuilder(endpoint, accessToken);
+  }
+};
+
+/**
+ * Returns lookup response using endpoint and accessToken
+ * @param {*} endpoint
+ * @param {*} accessToken
+ * @returns
+ */
+const userLookupResponseBuilder = async (endpoint, accessToken) => {
+  const requestOptions = {
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${accessToken}`
+    }
+  };
+  const lookupResponse = await httpGET(endpoint, requestOptions);
+  return processAxiosResponse(lookupResponse);
+};
+
+/**
+ * Returns 'Create User' payload
+ * @param {*} message
+ * @returns
+ */
+const createUserPayloadBuilder = message => {
+  const payload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.CREATE_USER.name]
+  );
+  const endpoint = CONFIG_CATEGORIES.CREATE_USER.endpoint;
+  const method = "POST";
+  validateEmailAndPhone(payload.email, payload.phone_number);
+  return { payload, endpoint, method };
+};
+
+/**
+ * Returns 'Update User' payload
+ * @param {*} message
+ * @returns
+ */
+const updateUserPayloadBuilder = message => {
+  const payload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.UPDATE_USER.name]
+  );
+  const endpoint = CONFIG_CATEGORIES.UPDATE_USER.endpoint;
+  const method = "PUT";
+  delete payload.external_id;
+  return { payload, endpoint, method };
+};
+
+/**
+ * Returns 'Creates Response' payload
+ * @param {*} message
+ * @returns
+ */
+const createResponsePayloadBuilder = message => {
+  const payload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.CREATE_RESPONSE.name]
+  );
+  const endpoint = CONFIG_CATEGORIES.CREATE_RESPONSE.endpoint;
+  const method = "POST";
+  validateScore(payload.score);
+  return { payload, endpoint, method };
+};
+
+/**
+ * Returns 'Creates Decline' payload
+ * @param {*} message
+ * @returns
+ */
+const createDeclinePayloadBuilder = message => {
+  const payload = constructPayload(
+    message,
+    MAPPING_CONFIG[CONFIG_CATEGORIES.CREATE_DECLINE.name]
+  );
+  const endpoint = CONFIG_CATEGORIES.CREATE_DECLINE.endpoint;
+  const method = "POST";
+  return { payload, endpoint, method };
+};
+
+/**
+ * Flattens properties field in payload
+ * e.g :- properties[name] = Demo User, end_user[properties][revenue_amount] = 5000
+ * ref: https://docs.wootric.com/api/#create-end-user
+ * ref: https://docs.wootric.com/api/#create-response
+ * @param {*} payload
+ * @param {*} destKey
+ */
+const flattenProperties = (payload, destKey) => {
+  if (payload.properties) {
+    let rawProperties = {};
+    Object.entries(payload.properties).forEach(([key, value]) => {
+      rawProperties[`${destKey}[${key}]`] = `${value}`;
+    });
+    return { ...rawProperties };
+  }
+};
+
+/**
+ * Formats identify payload
+ * ref: https://docs.wootric.com/api/#get-a-specific-end-user-by-phone-number
+ * @param {*} payload
```

Comment:
Lets add bit more contextual documentation for this function. Also the this function is stringifying dates but not formatting anything else for the entire payload. Contextual naming would be helpful?